modules/features/cu_article/cu_related_articles/cu_related_articles.module

```diff
@@ -0,0 +1,163 @@
+      $query->fields('cats', array('field_article_categories_tid'));
+      $query->condition('cats.field_article_categories_tid', $categories, 'in');
+      // Query for tags
+      $query->join("field_data_field_tags", "tags", "n.nid = tags.entity_id");
+      $query->fields('tags', array('field_tags_tid'));
+      $query->condition('tags.field_tags_tid', $tags, 'in');
+      // Get results
+      $results = $query->execute()->fetchCol();
+      // Count occurances of node id - more occurances = more related
+      $occurences = array_count_values($results);
+      // Sort by occurances
+      arsort($occurences);
+      // Add node ids to related articles array
+      foreach ($occurences as $key => $occurence) {
+        $related_articles[$key] = $key;
+      }
+    }
+
+    // Query #2 - Categories
+    if (!empty($categories)) {
+      $query = db_select('node', 'n');
+      $query->condition('n.type', 'article', '=');
+      $query->condition('n.status', 1, '=');
+      $query->fields('n', array('nid', 'title'));
+      // Don't include current article
+      $query->condition('n.nid', $current_nid, '!=');
+      // Query for categories
+      $query->join("field_data_field_article_categories", "cats", "n.nid = cats.entity_id");
+      $query->fields('cats', array('field_article_categories_tid'));
+      $query->condition('cats.field_article_categories_tid', $categories, 'in');
+      // Get results
+      $results = $query->execute()->fetchCol();
+      // Count occurances of node id - more occurances = more related
+      $occurences = array_count_values($results);
+      // Sort by occurances
+      arsort($occurences);
+      // Add node ids to related articles array
+      foreach ($occurences as $key => $occurence) {
+        $related_articles[$key] = $key;
+      }
+    }
+
+    // Query #3 - Tags
+    if (!empty($tags)) {
+      $query = db_select('node', 'n');
+      $query->condition('n.type', 'article', '=');
+      $query->condition('n.status', 1, '=');
+      $query->fields('n', array('nid', 'title'));
+      // Don't include current article
+      $query->condition('n.nid', $current_nid, '!=');
+      // Query for tags
+      $query->join("field_data_field_tags", "tags", "n.nid = tags.entity_id");
+      $query->fields('tags', array('field_tags_tid'));
+      $query->condition('tags.field_tags_tid', $tags, 'in');
+      // Get results
+      $results = $query->execute()->fetchCol();
+      // Count occurances of node id - more occurances = more related
+      $occurences = array_count_values($results);
+      // Sort by occurances
+      arsort($occurences);
+      // Add node ids to related articles array
+      foreach ($occurences as $key => $occurence) {
+        $related_articles[$key] = $key;
+      }
+    }
+    // If there are related articles, show the top 3.
+    if (!empty($related_articles)) {
+      $top_related = array_slice($related_articles, 0, 3, TRUE);
```

Comment:
could the 0 -3 part be 0 - variable_get('related_articles_list_number', 3) or something like that?